contents/tutorials/monitor-aws-bedrock-calls.md

```diff
@@ -0,0 +1,354 @@
+
+You must also ensure that you have properly configured your AWS credentials and region to use the AWS SDK for JavaScript. You can do this by calling `aws configure` using the [AWS CLI](https://docs.aws.amazon.com/cli/latest/userguide/getting-started-install.html).
+
+Once done, update the `BedrockRuntimeClient` initializer in `src/app/api/generate-recipe/route.js` to use your preferred AWS region:
+
+```js file=src/app/api/generate-recipe/route.js
+import {
+  BedrockRuntimeClient,
+  InvokeModelCommand,
+} from "@aws-sdk/client-bedrock-runtime";
+
+const client = new BedrockRuntimeClient({ region: "<YOUR_AWS_REGION>" }); // e.g. us-west-2
+
+// rest of the code
+```
+
+You'll also notice that we're using Meta's LLama 3.1 8B Instruct model. Make sure you have access to this model, or [request access](https://docs.aws.amazon.com/bedrock/latest/userguide/model-access.html) if you don't. Note that while this tutorial uses the Llama model, the concepts in this tutorial apply to all of Bedrock's [supported models](https://docs.aws.amazon.com/bedrock/latest/userguide/model-ids.html#model-ids-arns).
+
+![Requesting access to AWS Bedrock models](https://res.cloudinary.com/dmukukwp6/image/upload/v1723015919/posthog.com/contents/Screenshot_2024-08-07_at_8.31.28_AM.png)
+
+Run `npm run dev` and go to `http://localhost:3000` to everything in action.
+
+![Recipe builder LLM app](https://res.cloudinary.com/dmukukwp6/video/upload/v1723039836/posthog.com/contents/bedrock-sample-app.mp4)
+
+## 2. Add PostHog to your app
+
+With our app set up, it’s time to install and set up PostHog. We install the [PostHog Node](/docs/libraries/node) SDK to capture events in our API route. Run the following command in your terminal:
+
+```bash
+npm install posthog-node
+```
+
+Next, we initialize PostHog using our API key and host (You can find these in [your project settings](https://us.posthog.com/settings/project)). We also call `posthog.shutdown()` in a `finally` block to send any pending events before the serverless function shuts down. Add the below code to `src/app/api/generate-recipe/route.js`:
+
+```js file=src/app/api/generate-recipe/route.js
+// your existing imports
+
+import { PostHog } from 'posthog-node'; // import PostHog
+
+export async function POST(request) {
+  const posthog = new PostHog(
+    '<ph_project_api_key>',
+    {
+      host: '<ph_client_api_host>',
+    },
+  );
+
+  // rest of your code
+
+  try {
+    // existing code
+  } catch (error) {
+    // existing code
+  } finally {
+    // Call posthog.shutdown() to flush and send all pending events before the serverless function shuts down.
+    await posthog.shutdown();
+  }
+}
+```
+
+## 3. Capture events
+
+With our app set up, we can begin [capturing events](/docs/product-analytics/capture-events) with PostHog.
+
+### a. Successful requests
```

Comment:
```suggestion
### a. Successful requests

```

Reply:
Also, personally not sure the a., b., c. etc add very much, especially for the insights where they don't need to be ordered, just makes the sidebar/title slightly harder to read rather than help with ordering. But feel free to keep them if you feel strongly.

Reply:
thanks! will remove